src/main/java/be/seeseemelk/mockbukkit/entity/AreaEffectCloudMock.java

```diff
@@ -154,41 +153,31 @@ public <T> void setParticle(@NotNull Particle particle, @Nullable T data)
 	@Override
 	public void setBasePotionData(@Nullable PotionData data)
 	{
-		this.basePotionData = data;
+		this.potionType = data == null ? null : data.getType();
 	}
 
 	@Override
 	public @Nullable PotionData getBasePotionData()
 	{
-		return this.basePotionData;
+		return this.potionType == null ? null : new PotionData(this.potionType);
```

Comment:
I myself tried to implement these conversions, so I remember a bit about the paper implementation. You see, they introduced new PotionTypes which either are modifications of the strength of the potion effect, or the length of the potion effect. This data was previously stored in PotionData, where only the PotionType was the base effect of the potion. This is still the case for the now depricat4ed PoitoinData.

Please take a look in the paper class: `org.bukkit.craftbukkit.potion.CraftPotionUtil` most notably the `fromBukkit` and `toBukkit` functions.